Change UI/UX

subinasr opened the pull request:
- Add new image resources

AdityaKhatri commented on client/src/stylesheets/_base.scss:
```diff
@@ -75,3 +76,13 @@ body {
 }
+.container-block {
+    position: relative;
+    width: 1170px;
```
Don't use fixed widths.
If we go with 100% or 100vw, it will help us make responsive content.

AdityaKhatri commented on client/src/views/Home/Clients/styles.scss:
```diff
@@ -1,24 +1,40 @@
-            object-position: center center;
+.clients-section{
+    position: relative;
+    background-color: #555555;
```
Use variable

AdityaKhatri commented on client/src/views/Home/Clients/styles.scss:
```diff
@@ -1,24 +1,40 @@
+.clients-section{
+    position: relative;
+    background-color: #555555;
+    .overlay-block{
```
Add spacing before a block if there is a property above.

AdityaKhatri commented on client/src/views/Home/Expertise/index.js:
```diff
@@ -79,18 +81,20 @@ export default class Expertise extends React.PureComponent {
             <section
                 id="expertise"
-                className={className}
+                className={`${className} ${styles.expertiseSection}`}
```
Use '_cs()' function for '@togglecorp/fujs' library for className concatenation
```suggestion
                className={_cs(className, styles.expertiseSection)}
```

subinasr committed:
Change UI/UX

- Add new image resources

tnagorra commented on client/src/views/Home/Clients/index.js:
```diff
@@ -26,17 +27,19 @@ const rendererParams = (key, client) => ({ client });
-                title={client.name}
-            />
-        </a>
+        <div className={styles.clientInner}>
```
The rendererParams should be used to destructure `client`

tnagorra commented on client/src/views/Home/Clients/index.js:
```diff
@@ -58,18 +61,23 @@ export default class Clients extends React.PureComponent {
-                    rendererParams={rendererParams}
-                />
+                <div className={styles.overlayBlock}>
+                    <img src={clientsOverlay} alt="spike background" />
```
Multiple props should be written on separate line

tnagorra commented on client/src/views/Home/Footer/index.js:
```diff
@@ -0,0 +1,156 @@
+        socialIcon: 'fa fa-linkedin',
+    },
+];
+const rendererParams = (key, member) => ({ member });
```
The rendererParams should be used to destructure `member`

tnagorra commented on client/src/views/Home/Footer/index.js:
```diff
@@ -0,0 +1,156 @@
+                                title="Quick Links"
+                            />
+                            <ul>
+                                {footerMenuList.map(menuItem => (
```
Please use `List` component instead of `map`

tnagorra commented on client/src/views/Home/Footer/index.js:
```diff
@@ -0,0 +1,156 @@
+                                title="Follow US"
+                            />
+                            <ul className={styles.horizontalList}>
+                                {socialList.map(socialItem => (
```
Please use `List` component instead of `map`

tnagorra commented on client/src/views/Home/Services/index.js:
```diff
@@ -26,13 +26,15 @@ const rendererParams = (key, service) => ({ service });
-            {service.description}
-        </p>
+        <div className={styles.servicesInner}>
+            <img src={service.image} alt={service.title} />
```
Multiple props should be written on separate line

tnagorra commented on client/src/views/Home/Services/index.js:
```diff
@@ -50,21 +52,24 @@ export default class Services extends React.PureComponent {
+                        data={services}
+                        keyExtractor={keyExtractor}
+                        renderer={Service}
+                        rendererParams={rendererParams}
```
The rendererParams should be used to destructure `service`

tnagorra commented on client/src/views/Home/index.js:
```diff
@@ -163,19 +161,21 @@ export default class Home extends React.PureComponent {
-                />
+                <div className={styles.titleBlock}>
+                    <div className={styles.siteBranding}>
+                        <img src={logo} alt="Togglecorp logo" />
```
The rendererParams should be used to destructure client

tnagorra commented on client/src/views/Home/index.js:
```diff
@@ -163,19 +161,21 @@ export default class Home extends React.PureComponent {
+                </div>
+                <div className={styles.bottomBlock}>
+                    <AnchorLink
+                        href="#Home"
```
I think the href should be `#href`

tnagorra commented on client/src/views/Home/Team/index.js:
```diff
@@ -41,18 +42,23 @@ export default class Team extends React.PureComponent {
-                    rendererParams={rendererParams}
-                />
+                <div className={styles.overlayBlock}>
+                    <img src={teamOverlay} alt="abstract overlay" />
```
Multiple props should be written on separate line

tnagorra commented on client/src/views/Home/Footer/index.js:
```diff
@@ -0,0 +1,156 @@
+                className={`${className} ${styles.footerSection}`}
+            >
+                <div className={styles.overlayBlock}>
+                    <img src={footerOverlay} alt="abstract overlay" />
```
Multiple props should be written on separate line

tnagorra commented on client/src/views/Home/index.js:
```diff
@@ -135,21 +142,12 @@ export default class Home extends React.PureComponent {
-                    </p>
-                </div>
+            <div className={styles.backgroundOverlay}>
+                <img src={overlay} alt="Polka Dots" />
```
Multiple props should be written on separate line

tnagorra commented on client/src/views/Home/index.js:
```diff
@@ -163,19 +161,21 @@ export default class Home extends React.PureComponent {
+                    <AnchorLink
+                        href="#Home"
+                    >
+                        <img src={scroll} alt="Scroll Down" />
```
Multiple props should be written on separate line

tnagorra commented on client/src/views/Home/index.js:
```diff
@@ -135,21 +142,12 @@ export default class Home extends React.PureComponent {
+                <img src={overlay} alt="Polka Dots" />
             </div>
-            <div className={styles.rightBlock} >
+            <div className={styles.containerBlock}>
```
Please use `List` component instead of `map`

subinasr committed:
Change font

- Use List instead of Array.map
- Use CSS variables
- Fix indentation and spacing
- Change icons for darkmode

AdityaKhatri commented on client/src/views/Home/Expertise/index.js:
```diff
@@ -17,17 +18,19 @@ const tsRendererParams = (key, technologySection) => ({ technologySection });
 
 const Technology = ({ technology }) => (
     <li>
```
We can remove the "li" and achieve the same styling with less dom elements

AdityaKhatri commented on client/src/views/Home/Expertise/index.js:
```diff
@@ -17,17 +18,19 @@ const tsRendererParams = (key, technologySection) => ({ technologySection });
 
 const Technology = ({ technology }) => (
```
deconstruct technology and use url, image and name directly
```suggestion
const Technology = ({
    technology: {
        url,
        image,
        name,
    },
}) => (
```

AdityaKhatri commented on client/src/views/Home/Footer/index.js:
```diff
@@ -0,0 +1,168 @@
+
+const propTypes = {
+    className: PropTypes.string,
+    title: PropTypes.string.isRequired, // eslint-disable-line react/forbid-prop-types
```
Proptype string is valid so no eslint disabling

AdityaKhatri commented on client/src/views/Home/Footer/index.js:
```diff
@@ -0,0 +1,168 @@
+
+const socialKeyExtractor = social => social.id;
+const socialRendererParams = (key, social) => ({ social });
+const socialRenderer = ({ social }) => (
```
Could you deconstruct here?

AdityaKhatri commented on client/src/views/Home/Footer/index.js:
```diff
@@ -0,0 +1,168 @@
+];
+
+const footerMenuKeyExtractor = footerMenu => footerMenu.id;
+const footerMenuRendererParams = (key, footerMenu) => ({
```
Could you deconstruct footermenu here?

AdityaKhatri commented on client/src/views/Home/Services/index.js:
```diff
@@ -26,13 +27,15 @@ const rendererParams = (key, service) => ({ service });
 
 const Service = ({ service }) => (
```
Could you deconstruct service here?

AdityaKhatri commented on client/src/views/Home/Team/Member/index.js:
```diff
@@ -13,14 +16,19 @@ const keyExtractor = member => member.id;
+        <span className={_cs(
+            styles.link,
+            'fa',
+            `fa-${memberUrl.type.name}`,
```
memberUrl.type.name may result in name of undefined.
It would be safe if we write a validation as ((memberUrl || {}).type || {}).name

AdityaKhatri commented on client/src/views/Home/Team/Member/index.js:
```diff
@@ -13,14 +16,19 @@ const keyExtractor = member => member.id;
 const rendererParams = (key, memberUrl) => ({ memberUrl });
 
 const MemberUrl = ({ memberUrl }) => (
```
Could you deconstruct memberUrl here?

subinasr committed:
Class concatenation using fujs instead of template literals

AdityaKhatri commented on client/src/views/Home/Clients/index.js:
```diff
@@ -27,25 +27,32 @@ const defaultProps = {
+        name,
+    },
+}) => (
+    <div key={id} className={styles.client}>
```
Could you write the props in 2 separate lines? It will make it more readable

subinasr committed:
Deconstruct PropTypes

-Removed ESLint disabling
-Code Formatting and Indentation
-Responsive Layout fixed

tnagorra commented on client/public/index.html:
```diff
@@ -37,13 +37,13 @@
     <!-- End -->
 
-    <script>
+    <script>s
```
Looks like the `s` is misplaced

tnagorra commented on client/src/stylesheets/_base.scss:
```diff
@@ -75,3 +77,23 @@ body {
+        color: $color-accent-light;
+        font-weight: $font-weight-bold;
+
+        @media screen and (max-width: 720px) {
```
Shouldn't css variable be used instead of `720px`

tnagorra commented on client/src/views/Home/Footer/index.js:
```diff
@@ -0,0 +1,184 @@
+    } = social;
+    return { link, icon };
+};
+const socialRenderer = ({ link, icon }) => (
```
Maybe we can rename socialRenderer to 'SocialLink'

tnagorra commented on client/src/views/Home/Services/index.js:
```diff
@@ -22,21 +23,37 @@ const defaultProps = {
-            {service.description}
-        </p>
+const Service = ({ id, image, title, description }) => (
+    <div key={id} className={styles.service}>
```
Props should be placed on multiple lines

tnagorra commented on client/src/views/Home/Team/Member/index.js:
```diff
@@ -35,14 +52,19 @@ export default class Member extends React.PureComponent {
-                    <div className={styles.name}>{member.name}</div>
-                    <div className={styles.designation}>{member.designation}</div>
+                    <div className={styles.name}>
+                        {member.name}
```
We should deconstruct `member` props

subinasr committed:
Deconstructing in the rendererParams
- Style fixing
- Line separation for multiple props

subinasr committed:
Responsive device width variables updated

-classname starts with an uppercase
